src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -623,19 +635,16 @@ bool fx_muxer_t::resolve_sdk_dotnet_path(const pal::string_t& own_dir, pal::stri
             }
         }
     }
-    else
-    {
-        trace::verbose(_X("Failed to obtain current working dir"));
-    }
 
     std::vector<pal::string_t> hive_dir;
     pal::string_t local_dir;
     pal::string_t global_dir;
-    bool multilevel_lookup = multilevel_lookup_enabled();
+
+    bool multilevel_lookup = own_dir.empty() ? true : multilevel_lookup_enabled();
```

Comment:
own_dir is empty if and only if we're invoked via the new entry point that I am adding. Thus, any branch taken on own_dir.empty() is taken only when invoked via the new entry point. As such, this does not alter any existing behavior. If a user has set the environment variable to disable multilevel lookup, then they will still get the 1.x behavior of dotnet.exe invocations.

The only question is how should setting this setting should impact calls through the new entry point. I am arguing that it should not impact it at all. If it did, then desktop msbuild and VS would be broken in any environment with multilevel lookup disabled. I don't see how that could be desired behavior. There is no use for non-multilevel lookup when the executable (e.g. desktop `msbuild.exe`) doesn't ever carry the content it's seeking.

If you want, I can make the code easier to follow by adding an explicit `bool force_multilevel_lookup` instead of using own_dir.empty() to detect the "I'm not dotnet.exe, I have to go digging outside of my own directory to actually find anything" case.